pkg/controller/che/che_controller.go

```diff
@@ -1254,3 +1257,41 @@ func hasConsolelinkObject() bool {
 	}
 	return false
 }
+
+// GetFullCheServerImageLink evaluate full cheImage link(with repo and tag)
+// based on Checluster information and image defaults from env variables
+func GetFullCheServerImageLink(cr *orgv1.CheCluster) string {
+	cheImageRepo := cr.Spec.Server.CheImage
+
+	cheImageTag := util.GetValue(cr.Spec.Server.CheImageTag, deploy.DefaultCheVersion())
+
+	if len(cr.Spec.Server.CheImage) > 0 {
+		return cheImageRepo + ":" + cheImageTag
+	}
+
+	defaultCheServerImage := deploy.DefaultCheServerImage(cr)
+
+	if !strings.Contains(defaultCheServerImage, "@") {
```

Comment:
I would apply the following logic even when the `defaultCheServerImage` (coming from the env variable) contains a digest.
In other words, if the `defaultCheServerImage` contains a digest, but the `cheImageTag` is overridden in the `CheCluster` custom resource, then I would replace either the tag or the digest of the `defaultCheServerImage` by the `CheCluster` `cheImageTag` value

Reply:
@tolusha wdyt ?

Reply:
Thanks changing it @tolusha !